server/src/server.js

```diff
@@ -0,0 +1,166 @@
+          status: 400,
+          error: {
+            type: ERROR_INVALID_EVENT_ID,
+          },
+        });
+      } else {
+        const event = events[0];
+        if (
+          event === undefined ||
+          event === null ||
+          typeof event !== "object"
+        ) {
+          response.status(500).json({
+            status: 500,
+            error: {
+              type: ERROR_BAD_DB_INTERACTION,
+            },
+          });
+        } else {
+          request.event = event;
+          request.datastoreKey = key;
+          next();
+        }
+      }
+    })
+    .catch((err) => {
+      // TODO(ved): Potentially log to file instead of stderr
+      console.error(err)
+      response.status(500).json({
+        status: 500,
+        error: {
+          type: ERROR_BAD_DB_INTERACTION,
+        },
+      });
+    });
+}
+
+app.post("/create", async (_, response) => {
+  const key = datastore.key([KIND_EVENT]);
+  const result = await datastore.save({ key, data: { users: {} } });
+  response.send({
+    // Datastore automatically generates a unique id
+    // for the key associated with our entity. This is the only
+    // way to get it :(
+    // TODO: Check if the production data store server uses
+    // random numbers. The local emulator increments the ids,
+    // which makes them easily guessable. That's bad!
+    eventID: result[0].mutationResults[0].key.path[0].id,
+  });
+});
+
+app.post(`/:${URL_PARAM_EVENT_ID}`, getEvent, async (request, response) => {
+  const { body, datastoreKey: key, event } = request;
+  const location = JSON.parse(body.location);
+  const [lat, long] = location;
+  const { name } = body;
+  event.users = event.users || {};
+  const userInfo = event.users[request.session.userID] || {};
+  event.users[request.session.userID] = { ...userInfo, name, lat, long };
+  datastore
+    // Datastore attaches a "symbol" (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Symbol)
+    // to any entities returned from a query. We don't want to store this attached metadata back into the database
+    // so we remove it with Object.fromEntries
+    .save({ key, data: Object.fromEntries(Object.entries(event)) })
+    .then(() => {
+      response.json({ status: 200 });
+    })
+    .catch((err) => {
+      console.error(err)
+      response
+        .status(500)
+        .json({ status: 500, error: { type: ERROR_BAD_DB_INTERACTION } });
+    });
+});
+
+app.get(`/:${URL_PARAM_EVENT_ID}/details`, getEvent, async (request, response) => {
+  response.json({ status: 200, data: request.event });
+});
+
+app.get(`/:${URL_PARAM_EVENT_ID}/me`, getEvent, async (request, response) => {
+  const { event } = request;
+  const users = event.users || {};
+  const userInfo = users[request.session.userID] || {};
+  response.json({
+    status: 200,
+    data: userInfo,
+  });
+});
+
+const port = 3000;
+app.listen(port, () =>
+  console.log(`Server listening at http://localhost:${port}`)
+);
```

Comment:
Can you share a screenshot of the locally hosted server?

Reply:
I think a screenshot would not be super helpful. It might be best to pull this branch and try it out.

Reply:
alright